src/puppetlabs/puppetdb/utils.clj

```diff
@@ -310,6 +312,12 @@
                 #(update % 0 underscores->dashes)
                 m))
 
+(defn synthesize-body-str
+  [{:strs [command version certname payload]}]
+  (json/generate-string
+    {:command command :version (edn/read-string version)
+     :certname certname :payload (EncodedPayload. payload)}))
```

Comment:
I'd consider making a regular clojure fn for this so we don't need to mess around with the import. Probably not that important though.